public/app/features/dashboard-scene/settings/AnnotationsEditView.tsx

```diff
@@ -1,30 +1,107 @@
+  sceneGraph,
+} from '@grafana/scenes';
 import { Page } from 'app/core/components/Page/Page';
 
+import { DashboardScene } from '../scene/DashboardScene';
 import { NavToolbarActions } from '../scene/NavToolbarActions';
+import { dataLayersToAnnotations } from '../serialization/dataLayersToAnnotations';
 import { getDashboardSceneFor } from '../utils/utils';
 
+import { EditListViewSceneUrlSync } from './EditListViewSceneUrlSync';
+import { AnnotationSettingsList } from './annotations';
 import { DashboardEditView, DashboardEditViewState, useDashboardEditPageNav } from './utils';
 
-export interface AnnotationsEditViewState extends DashboardEditViewState {}
+export interface AnnotationsEditViewState extends DashboardEditViewState {
+  editIndex?: number | undefined;
+}
 
 export class AnnotationsEditView extends SceneObjectBase<AnnotationsEditViewState> implements DashboardEditView {
+  static Component = AnnotationsSettingsView;
+
   public getUrlKey(): string {
     return 'annotations';
   }
 
-  static Component = ({ model }: SceneComponentProps<AnnotationsEditView>) => {
-    const dashboard = getDashboardSceneFor(model);
-    const { navModel, pageNav } = useDashboardEditPageNav(dashboard, model.getUrlKey());
+  protected _urlSync = new EditListViewSceneUrlSync(this);
+
+  private get _dashboard(): DashboardScene {
+    return getDashboardSceneFor(this);
+  }
+
+  private get _dataLayers(): SceneDataLayerProvider[] {
+    return sceneGraph.getDataLayers(this._dashboard);
+  }
+
+  public getSceneDataLayers(): SceneDataLayers | undefined {
+    const data = sceneGraph.getData(this);
+
+    if (!(data instanceof SceneDataLayers)) {
+      return undefined;
+    }
+
+    return data;
+  }
+
+  public getAnnotationsLength(): number {
+    return this._dataLayers.filter((layer) => layer.topic === DataTopic.Annotations).length;
+  }
+
+  public getDashboard(): DashboardScene {
+    return this._dashboard;
+  }
 
-    return (
-      <Page navModel={navModel} pageNav={pageNav} layout={PageLayoutType.Standard}>
-        <NavToolbarActions dashboard={dashboard} />
-        <div>Annotations todo</div>
-      </Page>
-    );
+  public onNew = () => {
+    console.log('todo: onNew');
   };
+
+  public onEdit = (idx: number) => {
+    console.log('todo: onEdit');
+  };
+
+  public onMove = (idx: number, direction: number) => {
+    console.log('todo: onMove');
+  };
+
+  public onDelete = (idx: number) => {
+    console.log('todo: onDelete');
+  };
+}
+
+function AnnotationsSettingsView({ model }: SceneComponentProps<AnnotationsEditView>) {
+  const dashboard = model.getDashboard();
+  const sceneDataLayers = model.getSceneDataLayers();
+  const { navModel, pageNav } = useDashboardEditPageNav(dashboard, model.getUrlKey());
+  const { editIndex } = model.useState();
+
+  let annotations: AnnotationQuery[] = [];
+
+  if (sceneDataLayers) {
+    const { layers } = sceneDataLayers.useState();
+    annotations = dataLayersToAnnotations(layers);
+  }
```

Comment:
Can we make sure the getter throws if the data layers are not in place? Having hooks in conditions is something against hooks rules.

Reply:
Thank you! Fixed in the next [PR](https://github.com/grafana/grafana/pull/81361)